jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -102,9 +123,28 @@ public ReproducibleLayerBuilder addFiles(List<Path> sourceFiles, String extracti
    */
   public UnwrittenLayer build() throws IOException {
     List<TarArchiveEntry> filesystemEntries = new ArrayList<>();
+    List<LayerEntry> layerEntries = this.layerEntries.build();
+
+    // Adds the extraction paths (along with any parent directories) to explicitly set permissions
+    // for those directories.
+    LinkedHashSet<String> directoriesToAdd = new LinkedHashSet<>();
+    for (LayerEntry layerEntry : layerEntries) {
+      String extractionPath = layerEntry.getExtractionPath();
+      for (Path parentDirectory : getParentDirectories(Paths.get(extractionPath))) {
+        directoriesToAdd.add(parentDirectory.toString());
+      }
+    }
+    // We are using the current directory to act as the file input (since all directories are
```

Comment:
I'm not sure I understand this comment

Reply:
The `TarArchiveEntry` constructor can use a directory and sets up defaults like normalizing the path, normalizing the end separator (adds `/` if not present), and sets default permissions. It just needs to check `file.isDirectory`, so any `File` that is a directory will do.

Reply:
I think we could think about trying to find a better `TarArchiveEntry` or implement our own...